src/transformers/models/sam/image_processing_sam.py

```diff
@@ -934,7 +934,7 @@ def _generate_crop_boxes(
     cropped_images, point_grid_per_crop = _generate_crop_images(
         crop_boxes, image, points_grid, layer_idxs, target_size, original_size
     )
-
+    crop_boxes = np.array(crop_boxes)
```

Comment:
when it currently fails, what is the type of `crop_boxes`? Is it a torch tensor? If so, we can't rely on `np.array` unless we are 100% sure `crop_boxes` is never on cuda.

Reply:
Pre-processing should never be done on cuda no?